knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEngine.java

```diff
@@ -53,12 +56,18 @@ public ServiceEngine(Vertx vertx, TemplateEngineConfiguration serviceConfigurati
                     resp.subscribe(subscriber);
                     req.end();
                 });
-        return serviceResponse.flatMap(response ->
-                Observable.just(Buffer.buffer())
-                        .mergeWith(response.toObservable())
-                        .reduce(Buffer::appendBuffer))
-                .doOnNext(this::traceServiceCall)
-                .flatMap(buffer -> Observable.just(buffer.toJsonObject().getMap()));
+        return serviceResponse.flatMap(response -> Observable.just(Buffer.buffer())
+                .mergeWith(response.toObservable())
+                .reduce(Buffer::appendBuffer)
+                .flatMap(buffer -> {
+                    JsonObject jsonObject = buffer.toJsonObject();
+                    Map<String, String> responseMap = Maps.newHashMap();
+                    responseMap.put("statusCode", String.valueOf(response.statusCode()));
+                    jsonObject.getMap().put("_response", responseMap);
```

Comment:
We need put data directly to map `(buffer.toJsonObject().getMap())`
This `JsonObject` is provided by `vertex.io` and it keeps value type. 
In your example we will receive` key=(JsonObject)value` par and it causes a problem in handlebars parser.